src/components/pageControl/index.js

```diff
@@ -25,9 +27,27 @@ function getSizeStyle(size, enlargeActive, index, currentPage) {
  * @image: https://user-images.githubusercontent.com/33805983/34663655-76698110-f460-11e7-854b-243d27f66fec.png
  * @example: https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/PageControlScreen.js
  */
-export default class PageControl extends BaseComponent {
+export default class PageControl extends PureComponent {
   static displayName = 'PageControl';
   static propTypes = {
+    /**
+     * Limit the number of page indicators shown.
+     * enlargeActive prop is disabled in this state,
+     * while mediumSize and smallSize are enabled (and disabled when this is disabled).
+     * When set to true there will be maximum of 7 shown.
+     * Only relevant when numOfPages > 5.
+     */
+    limitShownPages: PropTypes.bool,
+    /**
+     * Size of page indicator to the sides of the regular one.
+     * Only relevant when limitShownPages is in effect.
+     */
+    mediumSize: PropTypes.number,
```

Comment:
Let's unify the mediumSize and smallSize prop into a single prop that can either accpet
- single number value (and we'll add a default behavior) 
- array of values up to 3 values for the `default', `medium` and `small`

Reply:
Changed